src/lib.rs

```diff
@@ -175,27 +175,23 @@ fn json_get(ctx: &Context, args: Vec<String>) -> RedisResult {
     let mut space = String::new();
     let mut newline = String::new();
     while let Ok(arg) = args.next_string() {
-        match arg.to_uppercase().as_str() {
-            "INDENT" => {
-                indent = args.next_string()?;
-            }
-            "NEWLINE" => {
-                newline = args.next_string()?;
-            }
-            "SPACE" => {
-                space = args.next_string()?;
-            }
-            "NOESCAPE" => {
-                // Silently ignore. Compatibility with ReJSON v1.0 which has this option. See #168
-                continue;
-            } // TODO add support
-            "FORMAT" => {
-                format = Format::from_str(args.next_string()?.as_str())?;
-            }
-            _ => {
-                paths.push(Path::new(arg));
-            }
-        };
+        // fast way to consider arg a path given none of
+        // the bellow subcommand placeholders is larger than 8
+        if arg.len() > 8 {
+            paths.push(Path::new(arg));
+        } else if arg.eq_ignore_ascii_case("INDENT") {
+            indent = args.next_string()?;
+        } else if arg.eq_ignore_ascii_case("NEWLINE") {
+            newline = args.next_string()?;
+        } else if arg.eq_ignore_ascii_case("SPACE") {
+            space = args.next_string()?;
+        } else if arg.eq_ignore_ascii_case("NOESCAPE") {
+            continue;
+        } else if arg.eq_ignore_ascii_case("FORMAT") {
+            format = Format::from_str(args.next_string()?.as_str())?;
+        } else {
+            paths.push(Path::new(arg));
+        }
```

Comment:
```suggestion
        match arg {
            arg if arg.eq_ignore_ascii_case("INDENT") => indent = args.next_string()?,
            arg if arg.eq_ignore_ascii_case("NEWLINE") => newline = args.next_string()?,
            arg if arg.eq_ignore_ascii_case("SPACE") => space = args.next_string()?,
            // Silently ignore. Compatibility with ReJSON v1.0 which has this option. See #168 TODO add support
            arg if arg.eq_ignore_ascii_case("NOESCAPE") => continue,
            arg if arg.eq_ignore_ascii_case("FORMAT") => {
                format = Format::from_str(args.next_string()?.as_str())?
            }
            _ => paths.push(Path::new(arg)),
        };
```

Reply:
@gkorland removing `if arg.len() > 8 {` makes us go from 150K ops/sec to ~= 145K ops/sec. 
Meaning solely relying on `eq_ignore_ascii_case` is a good improvement, but using the fast condition check + eq_ignore_ascii_case will be better.
I'll commit the suggestion but add the condition check above.